component/src/test/java/org/ballerinalang/mongodb/MongoDBActionsTest.java

```diff
@@ -193,6 +201,22 @@ public void testBatchInsert() throws Exception {
         }
     }
 
+    @Test(description = "Tests MongoDB replaces a single document.")
+    public void testReplaceOne() throws Exception {
+        BValue[] results = BRunUtil.invoke(result, "replaceOne");
+        Assert.assertEquals(((BInteger) results[0]).intValue(), 1,
+                "Only 1 record should have been updated");
+        Document document1 = (Document) mongoCollection.find(Document.parse("{\"age\":\"35\"}")).first();
+        FindIterable iterable = mongoCollection.find(Document.parse("{\"age\":\"35\"}"));
+        int size = 0;
+        for (Object value : iterable) {
+            size++;
+        }
+        Assert.assertEquals(size, 2, "Only 2 records should have been received");
+        Assert.assertEquals(document1.getString("name"), "Esther",
```

Comment:
In this case don't both documents get updated with the replacement document? In that case, shouldn't we verify whether both are updated?